src/main/java/WayofTime/bloodmagic/command/sub/SubCommandRitual.java

```diff
@@ -0,0 +1,158 @@
+package WayofTime.bloodmagic.command.sub;
+
+import WayofTime.bloodmagic.BloodMagic;
+import WayofTime.bloodmagic.ritual.Ritual;
+import WayofTime.bloodmagic.tile.TileMasterRitualStone;
+import WayofTime.bloodmagic.util.helper.RitualHelper;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.world.World;
+import net.minecraftforge.server.command.CommandTreeBase;
+import net.minecraftforge.server.command.CommandTreeHelp;
+
+public class SubCommandRitual extends CommandTreeBase {
+    public SubCommandRitual() {
+        addSubcommand(new RitualCreate());
+        addSubcommand(new RitualRemove());
+        addSubcommand(new RitualRepair());
+        addSubcommand(new CommandTreeHelp(this));
+    }
+
+    @Override
+    public String getName() {
+        return "ritual";
+    }
+
+    @Override
+    public String getUsage(ICommandSender sender) {
+        return null;
+    }
+
+    public TileMasterRitualStone getMRS(ICommandSender sender) {
+        BlockPos pos = sender.getPosition().down();
+        World world = sender.getEntityWorld();
+        TileEntity tile = world.getTileEntity(pos);
+        if (tile instanceof TileMasterRitualStone) {
+            return (TileMasterRitualStone) tile;
+        }
+        return null;
+    }
+
+    class RitualCreate extends CommandTreeBase {
+        public RitualCreate() {
+            for (Ritual i : BloodMagic.RITUAL_MANAGER.getRituals()) {
+                addSubcommand(new CommandTreeBase() {
```

Comment:
haha no

There will not be a command for every single ritual.

Reply:
The reason I did this is because it allows command completion via tab. If there's a downside to that, I'd like to know :P

Reply:
Did `getTabCompletions` stop existing or something?

Reply:
Nope, I didn't know it exists, I'll look into it.